modules/dnn/test/test_onnx_importer.cpp

```diff
@@ -481,6 +482,61 @@ TEST_P(Test_ONNX_nets, Shufflenet)
     testONNXModels("shufflenet", pb);
 }
 
+TEST_P(Test_ONNX_nets, Resnet34_kinetics)
+{
+    if (backend != DNN_BACKEND_INFERENCE_ENGINE || target != DNN_TARGET_CPU)
+        throw SkipTestException("Only DLIE backend on CPU is supported");
+
+    String onnxmodel = findDataFile("dnn/resnet-34_kinetics.onnx");
+    Mat image0 = imread(findDataFile("dnn/dog416.png"));
+    Mat image1 = imread(findDataFile("dnn/street.png"));
+
+    Mat ref0 = blobFromNPY(_tf("data/output_kinetics0.npy"));
+    Mat ref1 = blobFromNPY(_tf("data/output_kinetics1.npy"));
+
+    std::vector<Mat> images_0(16, image0);
+    std::vector<Mat> images_1(16, image1);
+    Mat blob0 = blobFromImages(images_0, 1.0, Size(112, 112), Scalar(114.7748, 107.7354, 99.4750), true, true);
+    Mat blob1 = blobFromImages(images_1, 1.0, Size(112, 112), Scalar(114.7748, 107.7354, 99.4750), true, true);
+
+    Net permute;
+    LayerParams lp;
+    int order[] = {1, 0, 2, 3};
+    lp.set("order", DictValue::arrayInt<int*>(&order[0], 4));
+    permute.addLayerToPrev("perm", "Permute", lp);
+
+    permute.setInput(blob0);
+    Mat input0 = permute.forward();
+
+    permute.setInput(blob1);
+    Mat input1 = permute.forward();
+
+    int dims[] = {1, 3, 16, 112, 112};
+    input0 = input0.reshape(0, 5, &dims[0]);
+    input1 = input1.reshape(0, 5, &dims[0]);
+
+    Net net = readNetFromONNX(onnxmodel);
+    ASSERT_FALSE(net.empty());
+    net.setPreferableBackend(backend);
+    net.setPreferableTarget(target);
+
+    checkBackend(&input0, &ref0);
+    net.setInput(input0);
+    Mat out = net.forward();
+
+    // output range [-5, 11]
+    float l1 = 0.0013;
+    float lInf = 0.009;
```

Comment:
Have you tested FP16 targets (OpenCL, Movidius)?

Reply:
3D layers are supported only on CPU

Reply:
So we need to skip all the non-CPU targets?

```
net.setPreferableTarget(target);
```

Reply:
This is not enough?
` if (backend != DNN_BACKEND_INFERENCE_ENGINE || target != DNN_TARGET_CPU)
        throw SkipTestException("Only DLIE backend on CPU is supported");`

Reply:
Sure, I missed it. Thanks!